src/relations/ldap.py

```diff
@@ -0,0 +1,115 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Defines ldap relation event handling methods."""
+
+import logging
+
+from ops import framework
+
+from literals import RELATION_VALUES
+from utils import log_event_handler
+
+logger = logging.getLogger(__name__)
+
+
+class LDAPRelationHandler(framework.Object):
+    """Client for ldap relations."""
+
+    def __init__(self, charm, relation_name="ldap"):
+        """Construct.
+
+        Args:
+            charm: The charm to attach the hooks to.
+            relation_name: The name of the relation defaults to ldap.
+        """
+        super().__init__(charm, "ldap")
+        self.charm = charm
+        self.relation_name = relation_name
+
+        # Handle database relation.
+        self.framework.observe(
+            charm.on[self.relation_name].relation_created,
+            self._on_relation_created,
+        )
+        self.framework.observe(
+            charm.on[self.relation_name].relation_changed,
+            self._on_relation_changed,
+        )
+        self.framework.observe(
+            charm.on[self.relation_name].relation_broken,
+            self._on_relation_broken,
+        )
+
+    @log_event_handler(logger)
+    def _on_relation_created(self, event):
+        """Handle ldap relation created.
+
+        Args:
+            event: The relation created event.
+        """
+        if self.charm.config["charm-function"] != "usersync":
+            return
+
+        if event.relation:
+            event.relation.data[self.charm.app].update({"user": "admin"})
```

Comment:
Is this user meant to be hard-coded?

Reply:
Yes, unfortunately because the ldap operator is very minimal it just creates an admin user and password with the "admin" name. As we'll be connecting to IS' LDAP (which isn't charmed in staging and production) I left this as is as this relation will become unused.